web/packages/shared/hooks/useAsync.test.ts

```diff
@@ -171,3 +171,26 @@ test('run does not update state after being re-run when the callback returns a r
     await secondRunPromise;
   });
 });
+
+test('error and statusText are set when the callback returns a rejected promise', async () => {
+  const expectedError = new Error('whoops');
+
+  const { result, waitFor } = renderHook(() =>
+    useAsync(() => Promise.reject(expectedError))
+  );
+
+  let runPromise: Promise<any>;
+  let [, run] = result.current;
+
+  act(() => {
+    runPromise = run();
+  });
+  await waitFor(() => result.current[0].status === 'error');
+
+  const attempt = result.current[0];
+  expect(attempt['error']).toEqual(expectedError);
```

Comment:
`toEqual` just compares recursively all properties of objects, so in our case it will only compare the error message, leaving the object type unchecked.

I think it would be better to just compare references of the objects: `(attempt['error'] === expectedError).toEqual(true)`. I wanted to propose using `toBe/toStrictEqual`, but then I read [this article](https://dev.to/thejaredwilcurt/why-you-should-never-use-tobe-in-jest-48ca).

Reply:
I think the article is a little bit too overzealous. In most dynamic languages I used, there is a distinction on the language level between two "entities" being comparable to each other vs two "entities" pointing to the same object in memory.

In 90% of cases I typically use the former, but the language and the testing frameworks typically have something for the latter as well. So I think `toBe` is completely fine to make sure that `attempt.error` is the same error that was returned from the callback.

And `toBe` also gives a nice error message if the error serializes to the same string but happens to refer to a different object in memory:

```
error and statusText are set when the callback returns a rejected promise

    expect(received).toBe(expected) // Object.is equality

    If it should pass with deep equality, replace "toBe" with "toStrictEqual"

    Expected: [Error: whoops]
    Received: serializes to the same string
```